### PR TITLE
FS2-2955: Update to use plotly 2.35.2

### PR DESCRIPTION
**Pull request by chschan**
Updates JS bundle to use the latest version (2.35.2). Note that the original R plotly library is only up to 2.11.1 so we have not synced from there.

Tests running on https://testlord.displayr.com:20344/Status

### CODE REVIEW

File: R/utils.R [hunk 1]
```diff
@@ -424,7 +424,7 @@ verify_attr_names <- function(p) {
   # some layout attributes (e.g., [x-y]axis can have trailing numbers)
   attrs_name_check(
     sub("[0-9]+$", "", names(p$x$layout)),
-    c(names(Schema$layout$layoutAttributes), c("barmode", "bargap", "bargroupgap", "mapType")),
+    c(names(Schema$layout$layoutAttributes), c("barmode", "bargap", "bargroupgap", "barcornerradius", "mapType")),
```
Comment (JustinCCYap) [inline R/utils.R:427]: Just curious how did you know to change this?

**Comment by JustinCCYap** [comment]
Would it be necessary to update the R plotly library as well or can the 2 be done separately?

File: R/utils.R (hunk 1 above)
Comment (chschan) [reply R/utils.R:427]: I changed it for bargroup gap a while ago as well. If you try to use barcornerradius you get an ugly warning. I set option(warning=2) to make it an error and checked the stack trace

Comment (JustinCCYap) [reply R/utils.R:427]: Nice. So that means we could get rounded corners just from this PR?

**Comment by chschan** [comment]
What do you mean the R plotly library? Our other flip packages depend on this one (e.g. https://github.com/Displayr/flipStandardCharts/blob/master/DESCRIPTION#L50), so they'll get this change if we merge this in

**Comment by chschan** [comment]
I am wondering if we should do a sync with plot/plotly.R but their JS bundle is so far behind, it doesn't seem worthwhile. A lot of the changes were with functions we don't use

**Comment by JustinCCYap** [comment]
I was using the term from your description. I meant merge in the commits from the original plotly repo that the fork is based on. So you think it is not worthwhile. I think we should document this perhaps in the README?

File: R/utils.R (hunk 1 above)
Comment (chschan) [reply R/utils.R:427]: Yep, rounded corners will work with this. Is there anything else we needed?

Comment (JustinCCYap) [reply R/utils.R:427]: Would you be able to test out the “between” option for the “layer” setting: https://www.notion.so/displayr/Our-plotly-version-is-old-which-means-that-some-features-are-unavailable-b20b3ebe92354cc583d599f1a78768af?pvs=4#c4bb2c738cd5406787016e71fccdbeae

**Comment by chschan** [comment]
The main thing we are missing is changes for compatibility with the updated ggplot2. I'm not familiar with it, but some users might have written some custom code to use it. I feel like we can leave it until someone complains about new features not working? (I feel that change will probably break someone else's code so I want to defer)

**Comment by JustinCCYap** [comment]
> The main thing we are missing is changes for compatibility with the updated ggplot2. I'm not familiar with it, but some users might have written some custom code to use it. I feel like we can leave it until someone complains about new features not working? (I feel that change will probably break someone else's code so I want to defer)

I don't understand much about how ggplot2 works with plotly, how do users write custom code to use it if we haven't updated our ploty repo?

**Review by JustinCCYap — APPROVED** [review]
LGTM. We'll need to figure out what other steps need to be done before merging this. E.g., testing ngvizes with the new plotly version and finding where else the plotly version is hardcoded. and determining the order of these steps.

**Pull request merged**